core/src/banking_stage/forwarder.rs

```diff
@@ -63,7 +63,7 @@ impl Forwarder {
 
         // get current root bank from bank_forks, use it to sanitize transaction and
         // load all accounts from address loader;
-        let current_bank = self.bank_forks.read().unwrap().root_bank();
+        let current_bank = self.bank_forks.read().unwrap().working_bank();
```

Comment:
This should stop us from filtering out transactions with blockhashes more recent than the current root bank.

I believe this is what caused the ~10 second latency increase when switching to confirmed commitment level for blockhashes. RPC would keep retrying sends and staked node would drop them until the confirmed blockhash became rooted.